Homework5/Routers/Routers.Tests/GraphTest.cs

```diff
@@ -0,0 +1,128 @@
+using NUnit.Framework;
+using Routers;
+
+using System;
+using System.IO;
+
+namespace Routers.Tests
+{
+    public class Tests
+    {
+        private string pathToInputFile = "../../../../Routers.Tests/TestInput.txt";
+
+        private void PrintDataToFile(string[] linesForFile)
+        {
+            var file = new StreamWriter(pathToInputFile);
+            foreach (var line in linesForFile)
+            {
+                file.WriteLine(line);
+            }
+            file.Close();
+        }
+
+        [SetUp]
+        public void CreatingFile()
+        {
+            FileStream output = File.Create(pathToInputFile);
+            output.Close();
+        }
+
+        [TestCase(new object[] { "4: " })]
+        [TestCase(new object[] { "1: 2 (w)" })]
+        [TestCase(new object[] { "1: r (2)" })]
+        [TestCase(new object[] { "q: 1 (5)" })]
+        [TestCase(new object[] { "3: 2 ()" })]
+        [TestCase(new object[] { "7: 1 {2}, 5 {4}" })]
+        [TestCase(new object[] { "7: 1 (2); 5 (4)" })]
+        public void CreatingGraphWithInvalidInputDataShallThrowExceprion(params string[] linesForFile)
+        {
+            PrintDataToFile(linesForFile);
+
+            bool testFail = true;
+            try
+            {
+                var a = new Graph(pathToInputFile);
+            }
+            catch (InvalidDataException)
+            {
+                testFail = false;
+            }
+            if (testFail)
+            {
+                Assert.Fail();
+            }
+            else
+            {
+                Assert.Pass();
+            }
```

Comment:
```suggestion
            Assert.Throws<InvalidDataException>(() => new Graph(pathToInputFile));
```
Если Assert.Throws на тот момент не умели, то 
```csharp
            if (testFail)
            {
                Assert.Fail();
            }
            else
            {
                Assert.Pass();
            }
```
можно было записать как
```csharp
Assert.False(testFail)
````
Да и вместо `var a` можно было `var _`